Marlin/Marlin_main.cpp

```diff
@@ -9483,28 +9483,35 @@ inline void gcode_M999() {
 }
 
 #if ENABLED(SWITCHING_EXTRUDER)
-  inline void move_extruder_servo(uint8_t e) {
-    #if (EXTRUDERS == 2)
-      const int angles[2] = SWITCHING_EXTRUDER_SERVO_ANGLES;
-      MOVE_SERVO(SWITCHING_EXTRUDER_SERVO_NR, angles[e]);
-    #else
-      const int angles[4] = SWITCHING_EXTRUDER_SERVO_ANGLES;
-      if (e < 2) MOVE_SERVO(SWITCHING_EXTRUDER_SERVO_NR, angles[e]);
-      else MOVE_SERVO(SWITCHING_EXTRUDER_E23_SERVO_NR, angles[e]);
-    #endif 
-    safe_delay(500);
+  inline void move_extruder_servo(const uint8_t e) {
+    stepper.synchronize();
+    constexpr int16_t angles[] = SWITCHING_EXTRUDER_SERVO_ANGLES;
+    static_assert(COUNT(angles) / 2 >= E_STEPPERS, "SWITCHING_EXTRUDER_SERVO_ANGLES needs 2 angles per servo.");
+
+    #if E_STEPPERS & 1
+      if (e < E_STEPPERS - 1)
+    #endif
```

Comment:
I think there is should be `EXTRUDERS `checking, because on each 2 extruders we have only one stepper.